kratos/processes/compute_nodal_gradient_process.cpp

```diff
@@ -50,9 +50,13 @@ template<bool THistorical>
 void ComputeNodalGradientProcess<THistorical>::ComputeElementalContributionsAndVolume() {
 
     // Auxiliar containers
-    Matrix DN_DX, J0, InvJ0;
-    Vector N, values;
-    double detJ0 = 0.0;
+    struct tls_type
+    {
+        Matrix DN_DX, J0, InvJ0;
+        Vector N, values;
+        double detJ0 = 0.0;
+    };
+    tls_type TLS = tls_type();
```

Comment:
```suggestion
    tls_type tls;
```
following the naming convention